numpy/add_newdocs.py

```diff
@@ -463,6 +463,65 @@ def luf(lamdaexpr, *args, **kwargs):
 
     """))
 
+add_newdoc('numpy.core', 'nested_iters',
+    """
+    Create a tuple of `nditer`s which each iterate over different axes of an
+    array. They are ordered. Advancing one will change the subsequent
+    iterators to point at its new element.
+
+    Parameters
+    ----------
+    op : ndarray or sequence of array_like
+        The array(s) to iterate over.
+
+    axes : list of list of int
+        Each item is used as an "op_axes" argument to an nditer
+
+    flags, op_flags, op_dtypes, order, casting, buffersize (optional)
+        See `nditer` parameters of the same name
+
+    Returns
+    -------
+    iters : tuple of nditer
+        An nditer for each item in `axes`, outermost first
+
+    See Also
+    --------
+    nditer
+
+    Examples
+    --------
+
+    Basic usage. Note how vals is a "flattened" version of
+    [a[:, 0, :], a[:, 1, 0], a[:, 2, :]] since we specified
+    the first iter's axes as [1]
+
+    >>> a = np.arange(12).reshape(2, 3, 2)
+    >>> i, j = np.nested_iters(a, [[1], [0, 2]], flags=["multi_index"])
+    >>> while not i.finished:
+    ...     print(i.multi_index)
+    ...     while not j.finished:
+    ...         print('', j.multi_index, j.value)
+    ...         j.iternext()
+    ....    i.iternext()
```

Comment:
I'm dumb, this works just fine in the more convenient `for` loop form

Reply:
Extra dot too

Reply:
ok, will update to `for` on the next commit

Reply:
thanks!